qiita_db/metadata_template/test/test_util.py

```diff
@@ -69,6 +71,31 @@ def test_load_template_to_dataframe(self):
         exp.index.name = 'sample_name'
         assert_frame_equal(obs, exp)
 
+    def test_load_template_to_dataframe_xlsx(self):
+        mfp = join(dirname(abspath(getfile(currentframe()))), 'support_files')
+
+        # test loading a qiimp file
+        fp = join(mfp, 'a_qiimp_wb.xlsx')
+        obs = qdb.metadata_template.util.load_template_to_dataframe(fp)
+        exp = pd.DataFrame.from_dict(EXP_QIIMP, dtype=str)
+        exp.index.name = 'sample_name'
+        assert_frame_equal(obs, exp)
+
+        # test loading an empty qiimp file
+        fp = join(mfp, 'empty_qiimp_wb.xlsx')
+        with self.assertRaises(qdb.exceptions.QiitaDBColumnError) as error:
+            qdb.metadata_template.util.load_template_to_dataframe(fp)
+        self.assertEqual(
+            str(error.exception), "The 'sample_name' column is missing from "
+            "your template, this file cannot be parsed.")
```

Comment:
This isn't a show-stopper, but I am concerned because I think the auto-generated exception is inaccurate ... an empty QIIMP template *will* contain a sample_name column, as shown below, and the tsv produced from it will also have include column--actually, the tsv will contain ONLY a row of column names :)  I fear that an error message indicating the sample_name column isn't there when it actually is will mislead users/future maintainers when they try to figure out why their upload failed.
<img width="985" alt="screen shot 2018-06-01 at 4 45 27 pm" src="https://user-images.githubusercontent.com/10677935/40867752-a43d5bd0-65bb-11e8-9198-f6d242278985.png">

Reply:
@AmandaBirmingham, good point. The reason this was happening was all columns are empty so all of them are dropped and the final dataframe was empty, and the first test is to check for validity is if sample_name column is there. Anyway, I added a new tests (empty) and now the error should make more sense.